fitbenchmarking/utils/checkpoint.py

```diff
@@ -225,81 +235,82 @@ def load(self):
                        dict,
                        dict[str, list[str]]]
         """
-        output: 'dict[str,list[FittingResult]]' = {}
-        unselected_minimizers: 'dict[str, list[str]]' = {}
-        failed_problems: 'dict[str, list[str]]' = {}
-
-        for f in [self.options.checkpoint_filename,
-                  os.path.join(self.options.results_dir,
-                               self.options.checkpoint_filename)]:
+        output: Dict[str, list[FittingResult]] = {}
```

Comment:
Why is this `Dict[str, list[...]]` and not `Dict[str, List[...]]` or `dict[str, list[...]]`?

I think all 3 are correct since 3.9 (https://peps.python.org/pep-0585/). Is it an automated ruff change (i.e. we don't know why) or one that you put in to fix something?

Reply:
Thinking about this a little more, taking out the quotes probably led to syntax errors in the 3.8 tests for the 3rd option since that pep only came in in 3.9

Reply:
I think that in python 3.8, we can't use built in types for type checking. Once we update the python to 3.9 we can use `dict[str, list[...]]`. That would be the correct way to do it. But I want to create another issue to make typing consistent throughout the repo.

Reply:
There is a GitHub action that failed because I updated it to `dict[str, list[...]]` I was trying to find it but that is why I have revert the typing changes.